### DISCUSSION TITLE
How to limit Invocation ID length in SignalR Callee

### DISCUSSION THREAD

**Discussion by dubchek** [dotnet/aspnetcore#22226] [General]
I need to limit Inocation ID length in SignalR Callee. And I found the following text in documentation (https://github.com/dotnet/aspnetcore/blob/master/src/SignalR/docs/specs/HubProtocol.md):

> NOTE: Invocation IDs are arbitrarily chosen by the Caller and the Callee is expected to use the same string in all response messages.
> Callees may establish reasonable limits on Invocation ID lengths and terminate the connection when an Invocation ID that is too long is received.

However I can't find any documentation how to change this limit or at least find it current value for the Callee. Could you please give me the information I am looking for?

**Comment by davidfowl** [comment]
It's not possible. Why do you need to limit it?

**Reply by dubchek** [reply]
I have an issue: If Invocation ID has more that 59 characters caller request is failed. At the moment I haven't details of the error. I will try to get them. But I will be grateful if you provide me with information about Invocation ID limitations in SignalR.

**Reply by BrennanConroy** [reply]
Are you sure you're thinking of the Invocation ID and not the connection ID?

The connection ID is part of the query string and if you have other things like large auth tokens in your query string then a server or proxy might reject the request by default.

The Invocation ID is part of the message sent over the transport, and a size of 59 (while very unlikely) is very small and shouldn't cause any issues.

**Reply by dubchek** [reply]
Sorry. But unfortunately ATM I haven't enough information to answer. As soon as I get it I will investigate and let you know. Thank you
